### PR TITLE
Clickable boards

### PR DESCRIPTION
**Pull request by paulvanwamelen**
Infrastructure for clickable boards and a working example for "squares-checkered".

### CODE REVIEW

File: src/renderers/homeworlds.ts [hunk 1]
```diff
@@ -26,7 +26,8 @@ export class HomeworldsRenderer extends RendererBase {
         super("homeworlds");
     }
 
-    public render(json: APRenderRep, draw: Svg, options: IRendererOptionsIn): void {
+    public render(json: APRenderRep, draw: Svg, boardClick: (row: number, col: number, piece: string) => void,
```
Comment (Perlkonig) [inline src/renderers/homeworlds.ts:29]: This makes `boardClick` a mandatory argument. Would this be better specified in the `IRenderOptions` interfaces and made optional?

File: src/index.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ const validate = ajv.compile(schema);
 interface IRenderOptions {
     divid: string;
     divelem?: HTMLElement;
+    boardClick?: (row: number, col: number, piece: string) => void;
```
Comment (Perlkonig) [inline src/index.ts:14]: I see you added it to the render options as optional, but isn't this overridden by the required argument added to `render()`?

File: src/RendererBase.ts [hunk 3]
```diff
@@ -34,7 +34,8 @@ export abstract class RendererBase {
         this.name = name;
     }
 
-    public abstract render(json: APRenderRep, draw: Svg, opts: IRendererOptionsIn): void;
+    public abstract render(json: APRenderRep, draw: Svg, boardClick: (row: number, col: number, piece: string) => void,
```
Comment (Perlkonig) [inline src/RendererBase.ts:37]: This makes `boardClick` a required parameter.

**Review by Perlkonig — CHANGES_REQUESTED** [review]
This looks great. I just have one concern. This makes `boardClick` a required argument to the `render` function. But you've also added it as optional in IRendererOptions. Does that not throw an error if you don't pass a `boardClick`? Can you make it so that it is always optional and confirm that everything else works if it is omitted?

File: src/renderers/homeworlds.ts (hunk 1 above)
Comment (paulvanwamelen) [reply src/renderers/homeworlds.ts:29]: Yes, it is mandatory on the extensions of the RenderBase class, but we only call the render function from index.ts and it says:
```export function render(json: APRenderRep, opts = {} as IRenderOptions)```
where the optional boardClick from IRenderOptions kicks in. It then does:
```
    let boardClick: (row: number, col: number, piece: string) => void = (row, col, piece) => undefined;
    if (("boardClick" in opts) && (opts.boardClick != null) ) {
        boardClick = opts.boardClick;
    }
```
before it calls `renderer.render` with this boardClick.
Point is, to the outside world it is optional (and I've tested that). Does that address your concern?

**Comment by Perlkonig** [comment]
Sorry for misunderstanding. Thanks again.

**Pull request merged**